common/src/main/java/com/wynntils/screens/skillpointloadouts/SkillPointLoadoutsScreen.java

```diff
@@ -0,0 +1,577 @@
+    private boolean hasSaveNameConflict = false;
+
+    private Pair<String, SavableSkillPointSet> selectedLoadout;
+    private WynntilsButton loadButton;
+    private WynntilsButton deleteButton;
+    private WynntilsButton convertButton;
+
+    private SkillPointLoadoutsScreen() {
+        super(Component.literal("Skill Point Loadouts Screen"));
+    }
+
+    public static Screen create() {
+        return new SkillPointLoadoutsScreen();
+    }
+
+    @Override
+    protected void doInit() {
+        super.doInit();
+        Models.SkillPoint.populateSkillPoints();
+
+        populateLoadouts();
+
+        saveNameInput = new TextInputBoxWidget(
+                (int) (dividedWidth * 35),
+                (int) (dividedHeight * 24),
+                (int) ((dividedWidth * 48) - (dividedWidth * 35)),
+                BUTTON_HEIGHT,
+                (x) -> {
+                    saveAssignedButton.active = !x.isBlank();
+                    saveBuildButton.active = !x.isBlank();
+                    hasSaveNameConflict = false;
+                    saveAssignedButtonConfirm = false;
+                    saveBuildButtonConfirm = false;
+                },
+                this,
+                saveNameInput);
+        this.addRenderableWidget(saveNameInput);
+
+        saveAssignedButton =
+                new WynntilsButton(
+                        (int) (dividedWidth * 49),
+                        (int) (dividedHeight * 24),
+                        (int) ((dividedWidth * 53) - (dividedWidth * 49)),
+                        BUTTON_HEIGHT,
+                        Component.translatable("screens.wynntils.skillPointLoadouts.save")) {
+                    @Override
+                    public void onPress() {
+                        String name = saveNameInput.getTextBoxInput();
+                        if (Models.SkillPoint.hasLoadout(name) && !saveAssignedButtonConfirm) {
+                            hasSaveNameConflict = true;
+                            saveAssignedButtonConfirm = true;
+                            this.setMessage(Component.translatable("screens.wynntils.skillPointLoadouts.confirm")
+                                    .withStyle(ChatFormatting.RED));
+                        } else {
+                            Models.SkillPoint.saveCurrentSkillPoints(name);
+                            populateLoadouts();
+                            setSelectedLoadout(new Pair<>(
+                                    name, Models.SkillPoint.getLoadouts().get(name)));
+                            saveNameInput.setTextBoxInput("");
+                            this.setMessage(Component.translatable("screens.wynntils.skillPointLoadouts.save"));
+                            // in case user pressed on both buttons
+                            saveBuildButton.setMessage(
+                                    Component.translatable("screens.wynntils.skillPointLoadouts.saveBuild"));
+                        }
```

Comment:
We've always just created new classes extending WynntilsButton, instead of using anonymous classes. It is a bit of boilerplate, but it improves this class' readability quite a bit.